external/CMakeLists.txt

```diff
@@ -27,26 +27,26 @@ function(pop_variable var)
 endfunction()
 
 if (DEFINED SPIRV-Headers_SOURCE_DIR)
-  # This allows flexible position of the SPIRV-Headers repo.
-  set(SPIRV_HEADER_DIR ${SPIRV-Headers_SOURCE_DIR})
+  add_subdirectory(${SPIRV-Headers_SOURCE_DIR})
+  set(SPIRV-Headers_INCLUDE_DIRS ${SPIRV-Headers_SOURCE_DIR}/include)
+elseif (IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/spirv-headers/)
+  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/spirv-headers/)
+  set(SPIRV-Headers_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/spirv-headers/include)
```

Comment:
The old implementation modified the parent scope when setting the include dir. I don't know if any projects rely on that or not. You can trying leaving it this way, but we may have to revert it again if it breaks something.